### ISSUE TITLE
Change (or drop) Statistics to package extension

### ISSUE THREAD

**Issue by thchr** [JuliaArrays/StaticArrays.jl#1171]
StaticArrays depends on Statistics which depends on SparseArrays. But SparseArrays is no longer in the sysimage, and this means it now contributes significantly (~70%) to the loading time of StaticArrays:
```jl
julia> @time_imports using StaticArrays
               ┌ 9.1 ms SparseArrays.CHOLMOD.__init__() 73.00% compilation time
    425.4 ms  SparseArrays 16.16% compilation time
      0.6 ms  Statistics
      0.7 ms  StaticArraysCore
    185.3 ms  StaticArrays
```

But Statistics is only used in order to extend (performance optimize?) `mean`, cf.:
https://github.com/JuliaArrays/StaticArrays.jl/blob/376efdbf3bbab34d624dfbee88a9ae83282f0718/src/StaticArrays.jl#L11
https://github.com/JuliaArrays/StaticArrays.jl/blob/376efdbf3bbab34d624dfbee88a9ae83282f0718/src/mapreduce.jl#L304-L310

So we could either use the new package extension functionality to load Statistics only conditionally - or we could entirely drop this (it's not entirely clear to me why it wouldn't be better to leave this to the [default `mean` implementation](https://github.com/JuliaStats/Statistics.jl/blob/bb7063d77a47e3fdc782d33b586d02c12908791e/src/Statistics.jl#L179) which e.g. dispatches to `mapreduce`)? 

Per discussion w/ @dkarrasch.

**Comment by thchr** [comment]
In fact, it seems that there is no noticeable performance advantage to using the extensions of `mean` from StaticArrays:

```jl
using StaticArrays, Statistics, BenchmarkTools
static_mean(A; kws...)    = mean(A; kws...)
static_mean(f, A; kws...) = mean(f, A; kws...)
base_mean(A; kws...)      = @invoke mean(A::AbstractArray; kws...)
base_mean(f, A; kws...)   = @invoke mean(f, A::AbstractArray; kws...)

S2 = @SMatrix rand(2,2)
S3 = @SMatrix rand(3,3)
M2 = @MMatrix rand(2,2)
M3 = @MMatrix rand(3,3)

@btime static_mean($S2) # 1.741 ns (0 allocations: 0 bytes)
@btime base_mean($S2)   # 1.985 ns (0 allocations: 0 bytes)

@btime static_mean($S3) # 2.004 ns (0 allocations: 0 bytes)
@btime base_mean($S3)   # 2.046 ns (0 allocations: 0 bytes)

@btime static_mean($M2) # 2.723 ns (0 allocations: 0 bytes)
@btime base_mean($M2)   # 2.478 ns (0 allocations: 0 bytes)

@btime static_mean($M3) # 2.727 ns (0 allocations: 0 bytes)
@btime base_mean($M3)   # 2.727 ns (0 allocations: 0 bytes)
```

Unfortunately, means over particular dimensions doesn't work for `SMatrix` since Statistic's `mean` doesn't anticipate that it might be impossible to `setindex` into a derived container:

```jl
julia> base_mean(S3; dims=1)
ERROR: setindex!(::SMatrix{1, 3, Float64, 3}, value, ::Int) is not defined.
 Hint: Use `MArray` or `SizedArray` to create a mutable static array
```

This is the [offending line](https://github.com/JuliaStats/Statistics.jl/blob/bb7063d77a47e3fdc782d33b586d02c12908791e/src/Statistics.jl#L191). It would be ideal to fix this on the Statistics.jl side and then just drop the extensions here, I think - but I cannot currently see how to do that neatly without sacrificing performance for mutable containers.

**Comment by thchr** [comment]
Fixed by #1172
